crates/common/fork_choice/lean/src/store.rs

```diff
@@ -728,3 +729,365 @@ impl Store {
+        (0..number_of_validators)
+            .map(|index| Validator {
+                public_key: PublicKey::from(&[0_u8; 52][..]),
+                index: index as u64,
+            })
+            .collect()
+    }
+
+    pub async fn sample_store(no_of_validators: usize) -> (Store, LeanState) {
+        let mut genesis_state =
+            LeanState::generate_genesis(0, Some(generate_default_validators(no_of_validators)));
+        let genesis_block = Block {
+            slot: 0,
+            proposer_index: 0,
+            parent_root: B256::ZERO,
+            state_root: genesis_state.tree_hash_root(),
+            body: BlockBody {
+                attestations: VariableList::empty(),
+            },
+        };
+        let genesis_block_hash = genesis_block.tree_hash_root();
+
+        let genesis_header = BlockHeader {
+            slot: genesis_block.slot,
+            proposer_index: genesis_block.proposer_index,
+            parent_root: genesis_block.parent_root,
+            state_root: genesis_block.state_root,
+            body_root: genesis_block.body.tree_hash_root(),
+        };
+
+        let signed_genesis_block = SignedBlockWithAttestation {
+            message: BlockWithAttestation {
+                block: genesis_block,
+                proposer_attestation: Attestation {
+                    validator_id: 0,
+                    data: AttestationData {
+                        slot: 0,
+                        head: Checkpoint::default(),
+                        target: Checkpoint::default(),
+                        source: Checkpoint::default(),
+                    },
+                },
+            },
+            signature: VariableList::default(),
+        };
+        let checkpoint = Checkpoint {
+            slot: 0,
+            root: genesis_block_hash,
+        };
+
+        set_lean_network_spec(LeanNetworkSpec::ephemery().into());
+        genesis_state.latest_block_header = genesis_header;
+        genesis_state.latest_finalized = checkpoint;
+        genesis_state.latest_justified = checkpoint;
+
+        let db = db_setup();
+
+        db.time_provider().insert(100).unwrap();
+        db.block_provider()
+            .insert(genesis_block_hash, signed_genesis_block.clone())
+            .unwrap();
+        db.latest_finalized_provider().insert(checkpoint).unwrap();
+        db.latest_justified_provider().insert(checkpoint).unwrap();
+        db.head_provider().insert(genesis_block_hash).unwrap();
+        db.safe_target_provider()
+            .insert(genesis_block_hash)
+            .unwrap();
+        db.state_provider()
+            .insert(genesis_block_hash, genesis_state.clone())
+            .unwrap();
+
+        (
+            Store {
+                store: Arc::new(tokio::sync::Mutex::new(db)),
+                network_state: Arc::new(NetworkState::new(checkpoint, checkpoint)),
+            },
+            genesis_state,
+        )
+    }
```

Comment:
can't we heavily reduce the code duplication here, moving `creates/common/chain/lean/genesis.rs` to this crate. Then use `pub fn setup_genesis`


Then use `Store::get_forkchoice_store()`, we should only need to add a new argument to it.

`time: Option<u64>` so we can optionally override the current value we have there